core/src/HIP/Kokkos_HIP_Abort.hpp

```diff
@@ -53,8 +53,14 @@
 namespace Kokkos {
 namespace Impl {
 
-[[noreturn]] __device__ __attribute__((noinline)) void hip_abort(
-    char const *msg) {
+[[noreturn]] __device__
+#ifndef KOKKOS_ENABLE_HIP_RELOCATABLE_DEVICE_CODE
+    __attribute__((noinline))
+#else
+    inline
+#endif
+    void
+    hip_abort(char const *msg) {
```

Comment:
Hmmm... can't you just do
```suggestion
[[noreturn]] __device__ __attribute__((noinline)) inline void hip_abort(
    char const *msg) {
```
Is there a problem declaring the function always `inline`?

Reply:
We decided not to inline because the compiler was generating poor code if you inline the `printf`. I don't know if the problem has been fixed or not.

Reply:
Sure, I didn't argue about `__attribute__((noinline))` but was asking about `inline`.